models/documentlistmodel.cpp

```diff
@@ -97,15 +98,32 @@ QHash< int, QByteArray > DocumentListModel::roleNames() const
     return d->roles;
 }
 
+void DocumentListModel::setAllItemsDirty(bool status)
+{
+    for(QList<DocumentListModelEntry>::iterator entry = d->entries.begin();
+        entry != d->entries.end(); entry++) {
+        entry->dirty = status;
+    }
+}
+
 void DocumentListModel::addItem(QString name, QString path, QString type, int size, QDateTime lastRead, QString mimeType)
 {
     // We sometimes get duplicate entries... and that's kind of silly.
-    foreach(const DocumentListModelEntry& entry, d->entries) {
-        if(entry.filePath == path)
+    for(QList<DocumentListModelEntry>::iterator entry = d->entries.begin();
+        entry != d->entries.end(); entry++) {
+        if(entry->filePath == path) {
+            entry->dirty = false;
+            entry->fileType = type;
+            entry->fileSize = size;
+            entry->fileRead = lastRead;
+            entry->mimeType = mimeType;
+            entry->documentClass = static_cast<DocumentClass>(mimeTypeToDocumentClass(mimeType));
```

Comment:
Why we need to compare all fields here ? path should be enough since file system can't have the same path twice, or we don't get full path here ?